src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,207 @@
+import { AfterViewInit, Directive, ElementRef, HostBinding, Input, OnChanges, OnDestroy, Renderer2, SimpleChanges } from '@angular/core';
+import { ColorService } from '../../services/color/color.service';
+import { ThemeColor } from '../../services/color/theme-color';
+import { ContrastService } from '../accessibility/contrast-ratio/contrast.service';
+
+export type BadgeVerticalPosition = 'above' | 'below';
+
+export type BadgeHorizontalPosition = 'before' | 'after';
+
+export type BadgeSize = 'small' | 'medium' | 'large';
+
+@Directive({
+    selector: '[uxBadge]',
+    exportAs: 'ux-badge',
+    host: {
+        class: 'ux-badge-container',
+        '[class.ux-badge-small]': 'badgeSize === "small"',
```

Comment:
Since we have the inline badge coming up, I think we'll have to make an adjustment here, The modifier classes should be on the `span.ux-badge` element rather than the parent element. This will allow us to use the badge styles on a standalone inline element.